test/raw_explain/test_raw_explanation.py

```diff
@@ -157,47 +157,98 @@ def test_get_local_raw_explanations_sparse_binary_classification(self, mimic_exp
+        feature_names = [str(i) for i in range(feature_map.shape[0])]
+
+        raw_eval_data = np.ones_like(iris[DatasetConstants.X_TRAIN])
+        global_raw_explanation = global_explanation.get_raw_explanation(
+            [feature_map],
+            raw_feature_names=feature_names[:feature_map.shape[0]],
+            eval_data=raw_eval_data)
+
+        assert np.array_equal(raw_eval_data, global_raw_explanation.eval_data)
+
+        self.validate_raw_explanation_classification(global_explanation, global_raw_explanation, feature_map,
+                                                     iris[DatasetConstants.CLASSES], feature_names,
+                                                     has_raw_eval_data=True)
 
-        assert np.array(global_raw_explanation.local_importance_values).shape[-1] == feature_map.shape[0]
+    def test_get_global_raw_explanations_regression_eval_data(self, boston, tabular_explainer):
+        model = create_sklearn_random_forest_regressor(boston[DatasetConstants.X_TRAIN],
+                                                       boston[DatasetConstants.Y_TRAIN])
 
-        assert global_raw_explanation.is_raw
-        assert not hasattr(global_raw_explanation, 'eval_data')
-        assert not global_raw_explanation.is_engineered
-        assert np.array(global_raw_explanation.global_importance_values).shape[-1] == feature_map.shape[0]
+        exp = tabular_explainer(model, boston[DatasetConstants.X_TRAIN], features=boston[DatasetConstants.FEATURES])
 
-    def validate_global_raw_explanation_classification(self, global_explanation, global_raw_explanation,
-                                                       feature_map, classes, feature_names, is_sparse=False):
+        global_explanation = exp.explain_global(boston[DatasetConstants.X_TEST])
         assert not global_explanation.is_raw
-        assert global_explanation.is_engineered
+        assert not global_explanation.is_engineered
+        num_engineered_feats = len(boston[DatasetConstants.FEATURES])
+        feature_map = np.eye(num_engineered_feats - 1, num_engineered_feats)
+
+        raw_eval_data = np.ones_like(boston[DatasetConstants.X_TRAIN])
+        global_raw_explanation = global_explanation.get_raw_explanation(
+            [feature_map],
+            eval_data=raw_eval_data)
+
+        assert np.array_equal(raw_eval_data, global_raw_explanation.eval_data)
+
+        self.validate_raw_explanation_regression(global_explanation, global_raw_explanation, feature_map,
+                                                 has_raw_eval_data=True)
+
+    def validate_raw_explanation_regression(self, eng_explanation, raw_explanation, feature_map,
+                                            has_eng_eval_data=True, has_raw_eval_data=False):
+        assert not eng_explanation.is_raw
+        assert hasattr(eng_explanation, 'eval_data') == has_eng_eval_data
+        assert eng_explanation.is_engineered
```

Comment:
Why are we validating engineered explanations if the method is meant to validate the raw explanations. The name is kind of confusing w.r.t what you are trying to do in the function. Could have another function for validating the engineered explanations..

Reply:
Updated the function name to not specify "raw"